polygeist/tools/cgeist/Lib/clang-mlir.cc

```diff
@@ -4859,8 +4900,10 @@ mlir::func::FuncOp MLIRASTConsumer::GetOrCreateMLIRFunction(
   if (!FD->isDefined(Def, /*checkforfriend*/ true))
     Def = FD;
 
-  if (functions.find(name) != functions.end()) {
-    auto function = functions[name];
+  auto &Context = F.context;
+
+  if (auto optFunction = getFunction(name, Context)) {
```

Comment:
Pls use actual types rather than auto (when the type is not clear).

Reply:
Done